src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,55 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] stack;
+    private int size;
 
+    public MateStack() {
+        this.stack = (T[]) new Object[DEFAULT_CAPACITY];
```

Comment:
`this` is redudant in this context
```suggestion
        stack = (T[]) new Object[DEFAULT_CAPACITY];
```

Reply:
changed